portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,121 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * When the user submits the form, Blobstore processes the file upload and then forwards the request
+ * to this servlet. This servlet can then process the request using the file URL we get from
+ * Blobstore.
+ */
+@WebServlet("/my-form-handler")
+public class FormHandlerServlet extends HttpServlet {
+  String BLOB_URL_PROPERTY = DataServlet.BLOB_URL_PROPERTY;
+  String COMMENT = DataServlet.COMMENT;
+  String CONTENT_PROPERTY = DataServlet.CONTENT_PROPERTY;
+  String TIMESTAMP_PROPERTY = DataServlet.TIMESTAMP_PROPERTY;
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the message entered by the user.
+    String message = request.getParameter("comment-input");
+
+    // Get the URL of the image that the user uploaded to Blobstore.
+    String imageUrl = getUploadedFileUrl(request, "image");
+  
+    // Create an entity with received comment message
+    long timestamp = System.currentTimeMillis();
+    Entity commentEntity = new Entity(COMMENT);
+    commentEntity.setProperty(CONTENT_PROPERTY, message);
+    commentEntity.setProperty(TIMESTAMP_PROPERTY,timestamp);
+    commentEntity.setProperty(BLOB_URL_PROPERTY, imageUrl);
+
+    // Used Datastore survice to store newly created comment entity
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    // Output some HTML that shows the data the user entered.
+    // A real codebase would probably store these in Datastore.
```

Comment:
Explanations associated with the code below: 
This is just a milestone from the blobstore step, which seems unnecessary since we have done the rest. I just realize that this whole displaying page can be removed since we fetch that back to our original index.html. 
Conclusion: 
I have removed comment and all unnecessary code.

Reply:
After testing, I added the redirecting feature to ensure user will not go to that uploading page for image storage